lib/src/metta/runner/modules/mod.rs

```diff
@@ -373,10 +389,26 @@ impl MettaMod {
 /// A ModuleLoader is responsible for loading a MeTTa module through the API.  Implementations of
 /// ModuleLoader can be used to define a module format or to supply programmatically defined modules
 pub trait ModuleLoader: std::fmt::Debug + Send + Sync {
-    /// A function to load the module my making MeTTa API calls.  This function will be called by
+    /// A function to load the module my making MeTTa API calls.  This function will be called
     /// as a downstream consequence of [Metta::load_module_at_path], [Metta::load_module_direct],
     /// [RunContext::load_module], or any other method that leads to the loading of modules
     fn load(&self, context: &mut RunContext) -> Result<(), String>;
+
+    /// Returns a data blob containing a given named resource belonging to a module
+    ///
+    /// The following values for `res_key` are commonly accepted:
+    /// * `module.metta`: The MeTTa code for the module in S-Expression format, if the module is
```

Comment:
1. I would suggest use name which doesn't look like a filename for the main code of the module. For instance we can call it `text`. The reason is that there can be different modules: ones which load source code from other files, ones which load source code from standard input. And I don't see why this main resource should have a name which is equal to the name of the file in one particular module format.

2. I see in code `"module.metta"` is frequently used as a string. I would suggest adding a constant for it. May be even some enum to keep all standard keys and allow non-standard ones:
```rs
enum ResKey<'a> {
  Text,
  Authors,
  Description,
  Other(&'a str),
}
```

Reply:
> I would suggest use name which doesn't look like a filename...

I think you're right.  When I conceived of the idea, I thought of them as files in a virtualized file-system, so it's not an accident they feel like file names.  In fact, my implementation started as a simple shim over files in the resources directory, but I felt that might be insecure so I locked it down more.

I now see that making the keys feel like files is going to lead to confusion when people think they can use any filename and it doesn't work.

> I would suggest adding a constant for it. May be even some enum to keep all standard keys and allow non-standard ones

That's a good idea.

Reply:
`ResourceKey` enum added in 3deba55